scripts/SHI/scpSHI_Jetz_complete.R

```diff
@@ -0,0 +1,396 @@
+      gdalcubes::filter_pixel(paste0("data >= ", min_elev)) |> select_bands("data")
+  }else{ # if only one value exist
+    if(condition == 3){ # if just maximum elevation available filter by that
+      cube_STRM_range <<- cube_STRM |>
+        gdalcubes::filter_pixel(paste0("data >= ", min_elev)) |> select_bands("data")
+    }
+    if(condition == 4){ # if just minimum elevation available filter by that
+      cube_STRM_range <<- cube_STRM |>
+        gdalcubes::filter_pixel(paste0("data <=", max_elev)) |> select_bands("data")
+    }
+  }
+  #convert to raster
+  r_STRM_range <<- cube_STRM_range  |> st_as_stars() |> terra::rast()
+
+  #resample to raster of SDM
+  r_STRM_range_res <<- terra::resample(r_STRM_range, r_suitability_map)
+  r_suitability_map <<- terra::mask(terra::crop(r_suitability_map,r_STRM_range_res),r_STRM_range_res) #Crop LC to suitability extent
+}
+)
+
+print("========== Map of suitable area generated ==========")
+
+#2.2 Load habitat preferences---------------------------------------------
+# df_IUCN_habitat_cat <- rl_habitats(sp,key = token)$result
+# 
+# #Load table with land cover equivalences need to be updated with Jung et al
+# df_IUCN_to_LC_categories <- read.csv(file.path(path_script,"SHI","IUCN_to_LC_categories.csv"),colClasses = "character") # PENDING PUT 0.5 TO MARGINAL HABITATS
+# df_IUCN_habitat_LC_cat <- left_join(df_IUCN_habitat_cat,df_IUCN_to_LC_categories, by="code")
+# LC_codes <- as.numeric(unique(df_IUCN_habitat_LC_cat$ESA_cod))
+
+#2.3 Hydrological features------------------------------------------------------
+
+
+#-------------------------------------------------------------------------------------------------------------------
+#3. Land Cover Values
+#-------------------------------------------------------------------------------------------------------------------
+#3.1 GFW data-------------------------------------------------------------------
+#forest base map
+cube_GFW_TC <-
+  load_cube(stac_path = "https://io.biodiversite-quebec.ca/stac/",
+            limit = 1000,
+            collections = c("gfw-treecover2000"),
+            bbox = sf_ext_crs,
+            srs.cube = srs_cube,
+            spatial.res = spat_res,
+            temporal.res = "P1Y",
+            t0 = "2000-01-01",
+            t1 = "2000-12-31",
+            resampling = "bilinear")
+
+cube_GFW_TC_range <- cube_GFW_TC |>
+  gdalcubes::filter_pixel(paste0("data >=", forest_threshold)) |>#filter to forest threshold 0-100
+  gdalcubes::apply_pixel("data/data") # turn into a forest presence map
+
+r_GFW_TC_range <- cube_to_raster(cube_GFW_TC_range , format="terra") # convert to raster format
+
+r_suitability_map_rescaled <- terra::resample(r_suitability_map,r_GFW_TC_range,method="mode") #Adjust scale of suitability map
+r_GFW_TC_range_mask <- r_GFW_TC_range |>
+  terra::classify(rcl=cbind(NA,0)) |> # turn NA to 0
+  terra::mask(r_suitability_map_rescaled) # crop to suitability map
```

Comment:
crop() first an then mask() to speed up the process? or it is not relevant here

Reply:
I didn't add it here because it is done for the cube data that has been loaded using the same extent, thus I thought it was not necessary